klyshko-mp-spdz/kii-run.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2022 - for information on the respective copyright owner
+# see the NOTICE file and/or the repository https://github.com/carbynestack/klyshko.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# Fail, if any command fails
+set -e
+
+# Setup offline executable command line arguments dictionary
+n=${KII_TUPLES_PER_JOB}
+pn=${KII_PLAYER_NUMBER}
+declare -A argsByType=(
+  ["BIT_GFP"]="--nbits 0,${n}"
+  ["BIT_GF2N"]="--nbits ${n},0"
+  ["INPUT_MASK_GFP"]="--ntriples 0,${n}"
+  ["INPUT_MASK_GF2N"]="--ntriples ${n},0"
+  ["INVERSE_TUPLE_GFP"]="--ninverses ${n}"
+  ["INVERSE_TUPLE_GF2N"]="--ninverses ${n}"
+  ["SQUARE_TUPLE_GFP"]="--nsquares 0,${n}"
+  ["SQUARE_TUPLE_GF2N"]="--nsquares ${n},0"
+  ["MULTIPLICATION_TRIPLE_GFP"]="--ntriples 0,${n}"
+  ["MULTIPLICATION_TRIPLE_GF2N"]="--ntriples ${n},0"
+)
+declare -A folderByType=(
+  ["BIT_GFP"]="2-p-128/Bits-p-P${pn}"
+  ["BIT_GF2N"]="2-2-40/Bits-2-P${pn}"
+  ["INPUT_MASK_GFP"]="2-p-128/Triples-p-P${pn}"
+  ["INPUT_MASK_GF2N"]="2-2-40/Triples-2-P${pn}"
+  ["INVERSE_TUPLE_GFP"]="2-p-128/Inverses-p-P${pn}"
+  ["INVERSE_TUPLE_GF2N"]="2-2-40/Inverses-2-P${pn}"
+  ["SQUARE_TUPLE_GFP"]="2-p-128/Squares-p-P${pn}"
+  ["SQUARE_TUPLE_GF2N"]="2-2-40/Squares-2-P${pn}"
+  ["MULTIPLICATION_TRIPLE_GFP"]="2-p-128/Triples-p-P${pn}"
+  ["MULTIPLICATION_TRIPLE_GF2N"]="2-2-40/Triples-2-P${pn}"
+)
+
+# Provide required parameters in MP-SPDZ "Player-Data" folder
+prime=$(cat /etc/kii/params/prime)
+declare fields=("p" "2")
+for f in "${fields[@]}"
+do
+
+  [[ "$f" = "p" ]] && bit_width="128" || bit_width="40"
+	folder="Player-Data/${KII_PLAYER_COUNT}-${f}-${bit_width}"
+	mkdir -p "${folder}"
+  echo "Providing parameters for field ${f}-${bit_width} in folder ${folder}"
```

Comment:
I agree that this could be improved by extracting the field type from the tuple type name and then only providing the parameters actually required. However, this would add a lot of complexity and wouldn't save a lot in terms of execution time. I added a disclaimer saying that parameters for both fields are added for reasons of simplicity. In that setting we have to compute the folder for both settings and can't just use the associative arrays. It prints always, true. But that's fine as it only says that it provides parameters in that folder and *not* tuples as you wrote.